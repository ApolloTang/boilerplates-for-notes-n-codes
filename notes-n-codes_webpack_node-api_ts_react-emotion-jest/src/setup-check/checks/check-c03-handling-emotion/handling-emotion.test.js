import {render} from '@testing-library/react';
import React from 'react';
import MyComponentWithEmotion from './handling-emotion.jsx';

import { createSerializer } from 'jest-emotion'
import * as emotion from '@emotion/core'

expect.addSnapshotSerializer(createSerializer(emotion)) // This is not working

test('It handle emotion', ()=>{
  const {container} = render(<MyComponentWithEmotion />);
  console.log('xxx container: ', container.innerHTML)
  expect(container.innerHTML).toMatchSnapshot();
});
