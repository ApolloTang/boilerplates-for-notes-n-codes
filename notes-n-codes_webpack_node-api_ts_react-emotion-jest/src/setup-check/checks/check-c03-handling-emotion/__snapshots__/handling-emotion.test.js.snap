// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It handle emotion 1`] = `"<hr><div class=\\"css-1ewwfb6-handling-emotion--myStyle\\">This is styled by @emotion, hover to change color.</div><button class=\\"css-m6ewyy-handling-emotion--Button ezrfpgc0\\">this is a styled.button</button><hr>"`;
